File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/CS/ComponentCS/Component/Component.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.EnterpriseServices;
using System.Reflection;

[assembly: ApplicationName("Component")]
[assembly: ApplicationAccessControl(false)]
[assembly: ApplicationActivation(ActivationOption.Server, SoapVRoot = "Component")]
[assembly: Description("Simple Serviced Component Sample")]
[assembly: AssemblyKeyFile("Component.snk")]


namespace Component
{
    public interface IMathOperation
    {
        int Add(int num1, int num2);
    }
    [EventTrackingEnabled(true)]
    [Description("Simple Serviced Component Sample")]

    public class Component : ServicedComponent, IMathOperation
    {
        public Component()
        {

        }

        public int Add(int num1, int num2)
        {
            return num1 + num2;
        }
    }
}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/VB/ComponentVB/Component/Component.vb
Imports System.EnterpriseServices
Imports System.Reflection

<Assembly: ApplicationName("Component")> 
<Assembly: ApplicationAccessControl(False)> 
<Assembly: ApplicationActivation(ActivationOption.Server, SoapVRoot:="Component")> 
<Assembly: Description("Simple Serviced Component Sample")> 
<Assembly: AssemblyKeyFile("Component.snk")> 

Public Interface IMathOperation
    Function Add(ByVal num1 As Integer, ByVal num2 As Integer) As Integer
End Interface
<EventTrackingEnabled(True), Description("Simple Serviced Component Sample")> _
Public Class Component
    Inherits ServicedComponent
    Implements IMathOperation
    Public Sub New()
    End Sub
    Public Function Add(ByVal num1 As Integer, ByVal num2 As Integer) As Integer Implements IMathOperation.Add
        Return num1 + num2
    End Function
End Class
